test/api/delegates.js

```diff
@@ -0,0 +1,356 @@
+    excessiveOffset: 10000
+};
+
+describe('Delegates API', function() {
+
+    /*Define functions for use within tests*/
+    function getActive(done) {
+        node.get('/api/delegates/getActive', done);
+    }
+
+    function getStandby(id, done) {
+        node.get('/api/delegates/getStandby?n=' + id, done);
+    }
+
+    function getLatestRegistrations(done) {
+        node.get('/api/delegates/getLatestRegistrations', done);
+    }
+
+    function getLastBlock(done) {
+        node.get('/api/delegates/getLastBlock', done);
+    }
+
+    function getLastBlocks(id1, id2, done) {
+        node.get('/api/delegates/getLastBlocks?publicKey=' + id1 + '&limit=' + id2, done);
+    }
+
+    function getSearch(id, done) {
+        node.get('/api/getSearch?q=' + id, done);
+    }
+
+    function getNextForgers(done) {
+        node.get('/api/delegates/getNextForgers', done);
+    }
+
+    function getDelegateProposals(done) {
+        node.get('/api/delegates/getDelegateProposals', done);
+    }
+
+    /*Testing functions */
+    function checkBlocks(id) {
+        for (var i = 0; i < id.length; i++) {
+            if (id[i + 1]) {
+                checkBlock(id[i]);
+            }
+        }
+    }
+
+    function checkBlock(id) {
+        node.expect(id).to.have.all.keys(
+            'delegate',
+            'totalForged',
+            'confirmations',
+            'blockSignature',
+            'generatorId',
+            'generatorPublicKey',
+            'payloadHash',
+            'payloadLength',
+            'reward',
+            'id',
+            'version',
+            'timestamp',
+            'height',
+            'previousBlock',
+            'numberOfTransactions',
+            'totalAmount',
+            'totalFee'
+        );
+    }
+
+    function checkDelegates(id) {
+        for (var i = 0; i < id.length; i++) {
+            if (id[i + 1]) {
+                checkDelegate(id[i]);
+            }
+        }
+    }
+
+    function checkDelegate(id) {
+        node.expect(id).to.contain.all.keys(
+            'productivity',
+            'username',
+            'address',
+            'publicKey',
+            'vote',
+            'producedblocks',
+            'missedblocks',
+            'rate',
+            'approval'
+        );
+    }
+
+    function checkDelegateProposals(id) {
+        for (var i = 0; i < id.length; i++) {
+            if (id[i + 1]) {
+                node.expect(id[i]).to.contain.all.keys(
+                    'topic',
+                    'name',
+                    'description'
+                );
+            }
+        }
+    }
+
+    function checkPublicKeys(id) {
+        for (var i = 0; i < id.length; i++) {
+            if (id[i + 1]) {
+                node.expect(id[i]).to.be.a.string;
+            }
+        }
+    }
+
+    function checkTransactions(id) {
+        for (var i = 0; i < id.length; i++) {
+            if (id[i + 1]) {
+                node.expect(id[i]).to.have.any.keys(
```

Comment:
test fails with all keys because knowlege only works on delegates and known accounts.